src/targets/objc/native.js

```diff
@@ -1,57 +1,76 @@
+
+        for (var i = 1, len = parameters.length; i < len; i++) {
+          code.push('[postData appendData:[@"&' + parameters[i].name + '=' + parameters[i].value + '" dataUsingEncoding:NSUTF8StringEncoding]];');
+        }
+        break;
+
+      case 'application/json':
+        try {
+          parameters = JSON.parse(source.postData.text);
+          code.push(objcHelpers.nsDictionaryBuilder('parameters', parameters, opts.indent));
+          code.push(null);
+          code.push('NSData *postData = [NSJSONSerialization dataWithJSONObject:parameters options:0 error:nil];')
+        } catch (e) {}
+        break;
+
+      case 'multipart/form-data':
+        break;
+
+      default:
+        code.push(null);
+        code.push('NSData *postData = [[NSData alloc] initWithData:[@"' + source.postData.text.replace(/"/g, '\\"') + '" dataUsingEncoding:NSUTF8StringEncoding]];');
+    }
+  }
+
   code.push(null);
-  code.push('NSURLSession *session = [NSURLSession sharedSession];');
-  code.push(null);
-  // Create request object
   code.push('NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:[NSURL URLWithString:@"' + source.fullUrl + '"]');
   code.push('                                                       cachePolicy:NSURLRequestUseProtocolCachePolicy');
   code.push('                                                   timeoutInterval:' + parseInt(opts.timeout, 10).toFixed(1) + '];');
   code.push('[request setHTTPMethod:@"' + source.method + '"];');
 
-  // Set headers
-  Object.keys(source.allHeaders).sort().map(function (key) {
-    code.push('[request setValue:@"' + source.allHeaders[key] + '" forHTTPHeaderField:@"' + key + '"];');
-  });
-
-  // Set request body
-  if (source.postData && (source.postData.params || source.postData.text)) {
-    code.push(null);
-
-    if (source.postData.mimeType === 'application/x-www-form-urlencoded' && source.postData.params) {
-      var params = source.postData.params;
-      code.push('NSMutableData *postData = [[NSMutableData alloc] initWithData:[@"' + params[0].name + '=' + params[0].value + '" dataUsingEncoding:NSUTF8StringEncoding]];');
-      for (var i = 1, len = params.length; i < len; i++) {
-        code.push('[postData appendData:[@"&' + params[i].name + '=' + params[i].value + '" dataUsingEncoding:NSUTF8StringEncoding]];');
-      }
-    } else if (source.postData.mimeType === 'application/json' && source.postData.text) {
-      code.push('NSData *postData = [[NSData alloc] initWithData:[@' +  JSON.stringify(source.postData.text) + ' dataUsingEncoding:NSUTF8StringEncoding]];');
-    } else if (source.postData.text) {
-      code.push('NSData *postData = [[NSData alloc] initWithData:[@"' + source.postData.text + '" dataUsingEncoding:NSUTF8StringEncoding]];');
-    }
+  if (Object.keys(source.headersObj).length) {
```

Comment:
you're using `source.allHeaders` best to keep these identical (use-case: no `source.headers`, but cookie was added to `source.allHeaders`)